### PR TITLE
provide the config in the constructor only.

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe PR

### CODE REVIEW

**Comment by baldersheim** [comment]
@havardpe Now this one should be suitable size for review.

File: fnet/src/vespa/fnet/transport.h [hunk 1]
```diff
@@ -17,6 +17,34 @@ class FNET_IServerAdapter;
...
+    TransportConfig & crypto(vespalib::CryptoEngine::SP crypto_in) {
+        _crypto = std::move(crypto_in);
+        return *this;
+    }
+    TransportConfig & config(const FNET_Config & config_in) {
+        _config = config_in;
+        return *this;
+    }
+    const FNET_Config & config() const { return _config; }
+    FNET_Config & config() { return _config; }
```
Comment (havardpe) [inline fnet/src/vespa/fnet/transport.h:40]: I suggest removing this

File: fnet/src/vespa/fnet/transport.h [hunk 2]
```diff
@@ -17,6 +17,34 @@ class FNET_IServerAdapter;
...
+    vespalib::CryptoEngine::SP crypto() const;
+    TransportConfig & resolver(vespalib::AsyncResolver::SP resolver_in) {
+        _resolver = std::move(resolver_in);
+        return *this;
+    }
+    TransportConfig & crypto(vespalib::CryptoEngine::SP crypto_in) {
+        _crypto = std::move(crypto_in);
+        return *this;
+    }
+    TransportConfig & config(const FNET_Config & config_in) {
```
Comment (havardpe) [inline fnet/src/vespa/fnet/transport.h:35]: I suggest removing this, replacing it with functions to set options directly

**Review by havardpe — COMMENTED** [review]
looks good overall. It seems you have not totally decided on how to set up internal fnet config values.

File: fnet/src/vespa/fnet/transport.h (hunk 1 above)
Comment (baldersheim) [reply fnet/src/vespa/fnet/transport.h:40]: This one was forgotten, thanks.

**Pull request merged**